runtime/CMakeLists.txt

```diff
@@ -18,6 +19,47 @@ add_custom_command(OUTPUT ${GENERATED_SYN_VIM}
     ${PROJECT_SOURCE_DIR}/src/nvim/eval.c
 )
 
+if(POLICY CMP0054)
+  cmake_policy(SET CMP0054 OLD)
+endif()
+
+file(GLOB PACKAGES ${PROJECT_SOURCE_DIR}/runtime/pack/dist/opt/*)
+
+set(GENERATED_PACKAGE_TAGS)
+foreach(PACKAGE ${PACKAGES})
+  get_filename_component(PACKNAME ${PACKAGE} NAME)
+  file(GLOB "${PACKNAME}_DOC_FILES" ${PACKAGE}/doc/*.txt)
+  if("${PACKNAME}_DOC_FILES")
+    file(MAKE_DIRECTORY ${GENERATED_PACKAGE_DIR}/${PACKNAME})
+    add_custom_target("${PACKNAME}-tags"
+      COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${PACKAGE} ${GENERATED_PACKAGE_DIR}/${PACKNAME}
+      COMMAND "${PROJECT_BINARY_DIR}/bin/nvim"
+        -u NONE -i NONE -e --headless -c "helptags doc" -c quit
+      DEPENDS
+        nvim
+      WORKING_DIRECTORY "${GENERATED_PACKAGE_DIR}/${PACKNAME}"
+    )
+
+    add_custom_command(OUTPUT "${GENERATED_PACKAGE_DIR}/${PACKNAME}/doc/tags"
+      DEPENDS
+        "${PACKNAME}-tags"
+      )
+
+    set("${PACKNAME}_DOC_NAMES")
+    foreach(DF "${${PACKNAME}_DOC_FILES}")
+      get_filename_component(F ${DF} NAME)
+      list(APPEND "${PACKNAME}_DOC_NAMES" ${GENERATED_PACKAGE_DIR}/${PACKNAME}/doc/${F})
+    endforeach()
+
+    install_helper(
+      FILES ${GENERATED_PACKAGE_DIR}/${PACKNAME}/doc/tags "${${PACKNAME}_DOC_NAMES}"
+      DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/nvim/runtime/pack/dist/opt/${PACKNAME}/doc)
+
+    list(APPEND GENERATED_PACKAGE_TAGS "${GENERATED_PACKAGE_DIR}/${PACKNAME}/doc/tags")
+  endif()
+endforeach()
```

Comment:
Yeah, I tried to model it after the existing helptags stuff.  Ideally, there'd be a way to turn this stuff into a function or something so there's not so much repetition.